### ISSUE TITLE
SDL_WINDOW_HIDDEN. BUG

### ISSUE THREAD

**Issue by baidwwy** [libsdl-org/SDL#7768]
```diff
+SDL_SetHint(SDL_HINT_ORIENTATIONS, "LandscapeLeft LandscapeRight");
+window = SDL_CreateWindow(NULL, 0, 0, 320, 480, SDL_WINDOW_FULLSCREEN | SDL_WINDOW_ALLOW_HIGHDPI|SDL_WINDOW_HIDDEN);
-window = SDL_CreateWindow(NULL, 0, 0, 320, 480, SDL_WINDOW_FULLSCREEN | SDL_WINDOW_ALLOW_HIGHDPI);

    renderer = SDL_CreateRenderer(window, -1, 0);

    SDL_GetWindowSize(window, &width, &height);
    SDL_RenderSetLogicalSize(renderer, width, height);
+SDL_ShowWindow(window);
```

SDL2 2.26.5

Missing display
![Simulator Screen Shot - iPhone SE (3rd generation) - 2023-06-01 at 21 01 27](https://github.com/libsdl-org/SDL/assets/6239458/d34c547a-72a8-448d-972c-f209bc02726c)

**Comment by slouken** [comment]
I'm not sure what the bug is here? Can you provide more explanation?

**Comment by baidwwy** [comment]
"If SDL_WINDOW_HIDDEN is used when creating and then SDL_ShowWindow is used, the screen will not be fully displayed."

**Comment by baidwwy** [comment]
"Please look at the smiley face in the screenshot, it is missing in the lower area."

**Comment by slouken** [comment]
This has been fixed for the 2.23.0 release, thanks!
